packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -22,6 +19,8 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
     name: 'asset-import-meta-url',
     async transform(code, id, ssr) {
       if (code.includes('new URL') && code.includes(`import.meta.url`)) {
+        const importMetaUrlRE =
+          /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*,\s*import\.meta\.url\s*\)/g
```

Comment:
I think we could reuse the regex to avoid re-parsing it? I saw this applied in other parts of the code base
```suggestion
        importMetaUrlRE.lastIndex = 0
```

Reply:
I've tried this approach. It didn't work out. I'm not familiar with the process of how
rollup handle transform phase. I guess different transform phases are interleaved.